credsweeper/scanner/scan_type/scan_type.py

```diff
@@ -89,43 +90,75 @@ def get_line_data(
-                    return None
-                candidate.line_data_list.append(aux_line_data)
-
-        return candidate
+            return candidates
+
+        line_data_list = cls.get_line_data_list(config=config,
+                                                target=target,
+                                                pattern=rule.patterns[0],
+                                                filters=rule.filters)
+
+        for line_data in line_data_list:
+            if config.exclude_values and line_data.value.strip() in config.exclude_values:
+                continue
+
+            candidate = Candidate([line_data], rule.patterns, rule.rule_name, rule.severity, config, rule.validations,
+                                  rule.use_ml)
+            # single pattern with multiple values means all the patterns must matched in target
+            if 1 < len(rule.patterns) and rule.rule_type in (RuleType.PATTERN, RuleType.KEYWORD):
+                # additional check whether all patterns match
+                if not cls._aux_scan(config, rule, target, candidate):
+                    # cannot find secondary values for the candidate
+                    continue
+            candidates.append(candidate)
+        return candidates
+
+    @classmethod
+    def _aux_scan(cls, config: Config, rule: Rule, target: AnalysisTarget, candidate: Candidate) -> bool:
+        """check for all secondary patterns and get nearest value"""
+        for pattern in rule.patterns[1:]:
+            line_data_list = cls.get_line_data_list(config=config, target=target, pattern=pattern, filters=rule.filters)
+
+            nearest_left: Optional[LineData] = None
+            nearest_right: Optional[LineData] = None
+            for line_data in line_data_list:
+                # standard filtering of values from config
+                if config.exclude_values and line_data.value.strip() in config.exclude_values:
+                    continue
+                # get values from both sides of primary value for speedup of one comparison
+                if line_data.value_end < candidate.line_data_list[0].value_start and (
+                        nearest_left is None or nearest_left.value_end < line_data.value_end):
+                    nearest_left = line_data
+                if line_data.value_start > candidate.line_data_list[0].value_end and (
+                        nearest_right is None or nearest_right.value_start > line_data.value_start):
+                    nearest_right = line_data
+
+            # append the values from both sides
+            if nearest_left:
+                candidate.line_data_list.append(nearest_left)
+            if nearest_right:
+                candidate.line_data_list.append(nearest_right)
+            # if none of both values - break the auxiliary scan with failure
+            if not nearest_left and not nearest_right:
+                return False
```

Comment:
Could you please explain why does `_aux_scan()` is needed?

I tried to understand the code but i can't find out the connection between `line_data`s with `candidate.line_data_list[0]`..
Why it needed to find `nearest_left` and `nearest_right`?

Reply:
It reduces possible possible wrong secondary values in a line. Only two values might be taken for primary value. From left and from right.